allennlp/interpret/attackers/hotflip.py

```diff
@@ -0,0 +1,203 @@
+                grad = grads[name_of_grad_input_field]
+                grads_mag = [numpy.sqrt(g.dot(g)) for g in grad]
+
+                # only flip a token once
+                for index in flipped:
+                    grads_mag[index] = -1
+
+                # we flip the token with highest gradient norm
+                index_of_token_to_flip = numpy.argmax(grads_mag)
+                if grads_mag[index_of_token_to_flip] == -1:
+                    break
+                flipped.append(index_of_token_to_flip)
+
+                # Get new token using taylor approximation
+                input_tokens = new_instances[0][name_of_input_field_to_attack]._indexed_tokens["tokens"]
+                original_id_of_token_to_flip = input_tokens[index_of_token_to_flip]
+                new_id_of_flipped_token = \
+                    first_order_taylor(grad[index_of_token_to_flip],
+                                       self.token_embedding.weight,
+                                       original_id_of_token_to_flip)
+                # flip token
+                new_instances[0][name_of_input_field_to_attack].tokens[index_of_token_to_flip] = \
+                    Token(self.vocab._index_to_token["tokens"][new_id_of_flipped_token])
+                new_instances[0].indexed = False
+
+                # Get model predictions on new_instances, and then label the instances
+                grads, outputs = self.predictor.get_gradients(new_instances) # predictions
+                for key in outputs:
+                    if isinstance(outputs[key], torch.Tensor):
+                        outputs[key] = outputs[key].detach().cpu().numpy().squeeze().squeeze()
+                    elif isinstance(outputs[key], list):
+                        outputs[key] = outputs[key][0]
+                # add labels to new_instances
+                self.predictor.predictions_to_labeled_instances(new_instances[0], outputs)
+
+                # if the prediction has changed, then stop
+                label_change = False
+                for field in fields_to_compare:
+                    if field in new_instances[0].fields:
+
+                        equal = new_instances[0][field].__eq__(fields_to_compare[field])
+                    else:
+                        equal = outputs[field] == fields_to_compare[field]
+                    if not equal:
+                        label_change = True
+                        break
+                # if the prediction has changed, we want to return the new answer
+                # for visualization in the demo.
+                if label_change:
+                    new_prediction = get_new_prediction(outputs)
+                    break
+
+            final_tokens.append(current_tokens)
+        return \
+            sanitize({"final": final_tokens, "original": original_tokens, "new_prediction": new_prediction})
+
+# Get the model's new prediction given its outputs
+def get_new_prediction(outputs):
```

Comment:
This is unfortunate.  Do we really need this?  It's already in the outputs.  Can't you just return the outputs in `sanitize` and let the demo find the answer that it needs?  The demo that called this knows what the task is, and where to find the new label.